src/signals/incident-management/containers/IncidentOverviewPage/components/List/index.test.js

```diff
@@ -153,7 +154,19 @@ describe('List', () => {
         element => element.textContent !== '-'
       );
 
-      expect(elementsWithTextContent).toHaveLength(2);
+      expect(elementsWithTextContent).toHaveLength(3);
+    });
+
+    it('should render an icon for each parent incident', () => {
+      const { container, rerender } = render(withContext(<List {...props} />));
+      const incidentList = [...props.incidents];
+
+      const parentsCount = incidentList.filter(incident => incident.has_children).length;
```

Comment:
I find this way too much code/logic for a unit test, making the test very error prone. I'd prefer a hard coded value for the count.

Reply:
I prefer no fixed values as then the tests are to be changed when there are some changing in the stucture/values of the test data.

Reply:
Because of the code changes you just made above, now you can see directly how many parents there are; exactly one, right? Then this is overkill.

Reply:
This construction allows me not to be concerned about the contents of the data but if the application does what it should. In this case will test that we render one image for each parent no matter how many parents there are.